src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -53,7 +53,7 @@ internal sealed partial class VisualStudioMetadataReferenceManager : IWorkspaceS
 
         internal VisualStudioMetadataReferenceManager(
             IServiceProvider serviceProvider,
-            ITemporaryStorageService temporaryStorageService)
+            TemporaryStorageServiceFactory.TemporaryStorageService temporaryStorageService)
```

Comment:
>TemporaryStorageServiceFactory.TemporaryStorageService [](http://example.com/codeflow?start=12&length=54)

nit: Would be nice to flip the nesting of these types so that we can avoid qualified names like these, i.e. 
```C#
internal class TemporaryStorageService
{
   [ExportWorkspaceServiceFactory(...)]
   internal class Factory { ... }
}
```